File: backend/app/schemas/product.py
from pydantic import BaseModel
from typing import Optional


class ProductBase(BaseModel):
    image: str
    name: str
    description: str
    price: float
    stock: int


class ProductCreate(ProductBase):
    pass


class ProductUpdate(BaseModel):
    name: Optional[str] = None
    price: Optional[int] = None
    stock: Optional[int] = None
    description: Optional[str] = None
    image: Optional[str] = None

    class Config:
        from_attributes = True


class Product(ProductBase):
    id: int

    class Config:
        from_attributes = True

File: frontend/src/api/interfaces/product.ts
import { PaginationObj } from './pagination'

export interface Product {
	id: number
	image: string
	name: string
	description: string
	price: number
	stock: number
}

export interface ProductForm {
	image: string
	name: string
	description: string
	price: string
	stock: string
	image_file: File | null
	new_image_path: string | null
	temp_image_path: string | null
}

export interface ProductError {
	image?: string
	name?: string
	description?: string
	price?: string
	stock?: string
	image_file?: string
}

export interface ProductQuery {
	search?: string
}

export interface PostData extends Omit<Product, 'id'> {}

export interface PaginatedProduct extends PaginationObj {
	items: Product[]
}
